seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/backend/BackendClient.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.doris.backend;
+
+import org.apache.seatunnel.shade.org.apache.thrift.TConfiguration;
+import org.apache.seatunnel.shade.org.apache.thrift.TException;
+import org.apache.seatunnel.shade.org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.seatunnel.shade.org.apache.thrift.protocol.TProtocol;
+import org.apache.seatunnel.shade.org.apache.thrift.transport.TSocket;
+import org.apache.seatunnel.shade.org.apache.thrift.transport.TTransport;
+import org.apache.seatunnel.shade.org.apache.thrift.transport.TTransportException;
+
+import org.apache.seatunnel.connectors.doris.config.DorisConfig;
+import org.apache.seatunnel.connectors.doris.config.DorisOptions;
+import org.apache.seatunnel.connectors.doris.exception.DorisConnectorErrorCode;
+import org.apache.seatunnel.connectors.doris.exception.DorisConnectorException;
+import org.apache.seatunnel.connectors.doris.source.serialization.Routing;
+import org.apache.seatunnel.connectors.doris.util.ErrorMessages;
+
+import org.apache.doris.sdk.thrift.TDorisExternalService;
+import org.apache.doris.sdk.thrift.TScanBatchResult;
+import org.apache.doris.sdk.thrift.TScanCloseParams;
+import org.apache.doris.sdk.thrift.TScanCloseResult;
+import org.apache.doris.sdk.thrift.TScanNextBatchParams;
+import org.apache.doris.sdk.thrift.TScanOpenParams;
+import org.apache.doris.sdk.thrift.TScanOpenResult;
+import org.apache.doris.sdk.thrift.TStatusCode;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class BackendClient {
+
+    private Routing routing;
+
+    private TDorisExternalService.Client client;
+    private TTransport transport;
+
+    private boolean isConnected = false;
+    private final int retries;
+    private final int socketTimeout;
+    private final int connectTimeout;
+
+    public BackendClient(Routing routing, DorisConfig readOptions) {
+        this.routing = routing;
+        this.connectTimeout =
```

Comment:
From `DorisConfig` we can know `readOptions.getRequestConnectTimeoutMs()` will never return null.

DorisConfig.java
```
dorisConfig.setRequestReadTimeoutMs(config.get(DORIS_REQUEST_CONNECT_TIMEOUT_MS));
```

ReadOnlyConfig.java
```
public <T> T get(Option<T> option) {
        return getOptional(option).orElseGet(option::defaultValue);
    }
```